server/form.php

```diff
@@ -5,20 +5,30 @@
 	$retrieve = $_GET['retrieve'];
 
 	if ($retrieve == "questions") {
-		// TODO make this handle multiple locations, if necessary
-		$questionStatement = $conn->prepare('SELECT questionId, string, inputType, placeholder, tag, subheading, required, archived FROM Question
-			WHERE (archived != true)
+		$subheadings = [];
+		if (isset($_GET['subheadings'])) {
+			$getSubheadings = $_GET['subheadings'];
+			if (is_array($getSubheadings)) {
+				$subheadings = $getSubheadings;
+			} else {
+				$subheadings[] = $getSubheadings;
+			}
+		}
+
+		$questionStatement = $conn->prepare('SELECT questionId, string, tag, required, archived, inputType, placeholder, subheading, validationType, hint, errorMessage FROM Question q
+			JOIN QuestionInformation qi on qi.questionInformationId = q.questionInformationId
+      WHERE (archived != true) and (qi.subheading = ?)
 			ORDER BY subheading');
-		$questionStatement->execute();
-		$results = $questionStatement->fetchAll();
 
 		$questions = [];
-		foreach ($results as $result) {
-			$result["errorMessage"] = "Sample error message.";
-			$result["validationType"] = "email";
-			$result["hint"] = "(Sample Hint)";
+		foreach ($subheadings as $subheading) {
+			$questionStatement->execute(array($subheading));
+
+			$results = $questionStatement->fetchAll();
 
-			$questions[] = $result;
+			foreach ($results as $result) {
```

Comment:
Consider:
```php
$results = $questionStatement->fetchAll();
$questions = array_merge($questions, $results);
```

Alternatively:
```php
$questions = array_merge($questions, $questionStatement->fetchAll());
```


[Here](http://php.net/manual/en/function.array-merge.php) is the documentation for `array_merge()`.
Just some quality of life code cleanup :P